packages/admin/components/composed/ordering/EntityOrderingAddForm/EntityOrderingAddForm.tsx

```diff
@@ -45,16 +52,43 @@ export default function EntityOrderingAddForm({
         .map((node) => node.node.identifier);
   };
 
+  const getOrder = (
+    property: string,
+    direction: string
+  ): OrderDefinitionInput[] => {
+    const directionValue = direction === "asc" ? "ASCENDING" : "DESCENDING";
+
+    interface PathOptions extends Record<string, string> {
+      name: string;
+      updatedAt: string;
+      published: string;
+    }
+    const pathOptions: PathOptions = {
+      name: "entity.title",
+      updatedAt: "entity.updated_at",
+      published: "entity.published_on",
+    };
+    const path = pathOptions[property];
+
+    return [{ path: path, direction: directionValue }];
+  };
+
   const toVariables = useToVariables<EntityOrderingAddFormMutation, Fields>(
     (data) => ({
       input: {
-        ...data,
         entityId: entity?.id || "",
-        /* placeholder */
-        identifier: data?.name?.toLowerCase() || "",
+        name: data.name,
+        /* Identifier should be `${convertToSlug(data.name)}-${uid} but the api errors if non-alpha characters are included.` */
+        identifier: data.name
+          ? data.name.toLowerCase().split(" ").join("")
+          : "",
         filter: {
-          schemas: getFilter(data.directItems, data.directCollections),
+          schemas: getFilter(data.items, data.collections),
         },
+        select: {
+          direct: data.collections !== "NONE" ? data.collections : data.items,
+        },
+        order: getOrder(data.sortby.split(",")[0], data.sortby.split(",")[1]),
```

Comment:
I don't love this. Is there a better way?

Reply:
I would pass sortby to getOrder, then do the splitting in the getOrder function:
```
const getOrder = (sortby: string) => {
   [property, direction] = sortby.split(",");
   ...
}
...
order: getOrder(data.sortby)
```
You might need to make some adjustments if order can be null or undefined, but I think that's pretty close.